pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -177,7 +177,7 @@ func (i *CommandInitOption) Complete() error {
 		}
 	}
 
-	return nil
+	return os.MkdirAll(i.KarmadaDataPath, 0755)
```

Comment:
Can you remind me why just removes the `KarmadaDataPath` and create it again?

Reply:
based on this PR: https://github.com/karmada-io/karmada/pull/1455
```shell
When the current host uses init to install karmada multiple times, the installation fails because the data directory of karmada is not empty.
```
After remove it, it's not create it again then it will has error here: https://github.com/karmada-io/karmada/blob/ab889889faa84b5a1c1ad89241862b642b3cfb4c/pkg/karmadactl/cmdinit/kubernetes/deploy.go#L252

Reply:
I just thought about that it is better not to remove whole directory, if users set a important existed directory, the command will remove it, which is dangerous. Like `kubeamd reset`, it only remove the files that created by `kubeadm init`. Perhaps, we can enhence it in a new PR.

For this pr, I think it is fine.